crates/bevy_diagnostic/src/diagnostic.rs

```diff
@@ -44,16 +54,16 @@ pub struct Diagnostic {
 }
 
 impl Diagnostic {
-    /// Add a new value as a [`DiagnosticMeasurement`]. Its timestamp will be [`Instant::now`].
-    pub fn add_measurement(&mut self, value: f64) {
-        let time = Instant::now();
+    /// Add a new value as a [`DiagnosticMeasurement`].
+    pub fn add_measurement(&mut self, value: impl Into<DiagnosticMeasurement>) {
```

Comment:
I'd prefer if this took a `DiagnosticMeasurement` over an `Into<DiagnosticMeasurement>`. Or even be made private. It's breaking, but using this method directly is footgunny, so better hint through the type that you aren't supposed to add the measurement through this API.

Reply:
I removed the `Into<>` and just use the `DiagnosticMeasurement` directly. I didn't make it private just because it seems like some people _might_ be using the `Diagnostic` struct without `Diagnostics` and I erred on the side of reducing API changes. I'm happy to change it to private if preferred.

Reply:
I trust your judgment! I think It's fine to leave it public, especially with the new doc infos you added.